app/components/generic-chart.js

```diff
@@ -0,0 +1,293 @@
+/* global c3 */
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+
+    classNames: ['chart'],
+
+    sourcesList: Ember.computed('data', function() {
+      if(this.get('chartType')=='donut-chart'){
+          return this.get('data').map(({ key, doc_count }) => [key, doc_count]);
+      }
+      else if(this.get('chartType')=='bar-chart'){
+          return this.get('data').map(({ key, doc_count }) => [key, doc_count]).slice(0, 10);
```

Comment:
Can move this differentiation to the query; so there wouldn't be any cases in this component?